packages/web/server/api/edge-routers/concentrated-liquidity-router.ts

```diff
@@ -0,0 +1,110 @@
+import { Dec, PricePretty, RatePretty } from "@keplr-wallet/unit";
+import { z } from "zod";
+
+import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
+import { mapRawAssetsToCoinPretty } from "~/server/queries/complex/assets";
+import { DEFAULT_VS_CURRENCY } from "~/server/queries/complex/assets/config";
+import {
+  getTotalClaimableRewards,
+  getTotalEarned,
+  normalizePositions,
+} from "~/server/queries/complex/concentrated-liquidity";
+import { UserOsmoAddressSchema } from "~/server/queries/complex/parameter-types";
+import { queryPositionPerformance } from "~/server/queries/imperator";
+import {
+  queryCLPosition,
+  queryCLPositions,
+} from "~/server/queries/osmosis/concentratedliquidity";
+import { sort } from "~/utils/sort";
+
+export const concentratedLiquidityRouter = createTRPCRouter({
+  getUserPositions: publicProcedure
+    .input(
+      z
+        .object({
+          sortDirection: z.enum(["asc", "desc"]).default("desc"),
+        })
+        .merge(UserOsmoAddressSchema.required())
+    )
+    .query(async ({ input: { userOsmoAddress, sortDirection } }) => {
+      const { positions: rawPositions } = await queryCLPositions({
+        bech32Address: userOsmoAddress,
+      });
+
+      const result = await normalizePositions({
+        positions: rawPositions,
+        userOsmoAddress,
+      });
+      return sort(result, "joinTime", sortDirection);
+    }),
+
+  getPositionPerformance: publicProcedure
+    .input(
+      z.object({
+        positionId: z.string(),
+      })
+    )
+    .query(async ({ input: { positionId } }) => {
+      const [position, performance] = await Promise.all([
+        queryCLPosition({ id: positionId }),
+        queryPositionPerformance({ positionId }),
+      ]);
```

Comment:
What is the benefit of wrapping in promise all? Does it send the requests simultaneously?

Reply:
It resolves promises concurrently. It's faster than if we use plain await